R/utils.R

```diff
@@ -493,6 +493,30 @@ shinyCallingHandlers <- function(expr) {
   )
 }
 
+shinyUserErrorUnhandled <- function(error, handler = NULL) {
+  if (is.null(handler)) {
+    handler <- getShinyOption(
+      "shiny.error.unhandled",
+      getOption("shiny.error.unhandled", NULL)
+    )
```

Comment:
This bit allows users to set the `shiny.error.unhandled` option inside a server function to have it be scoped to the session. Using an option in this case would cause the handler to fall out of scope before it's called.